rfcs/text/0000-dss-deletion-daemon.md

```diff
@@ -0,0 +1,268 @@
+database or index.
+
+### Derived data
+
+The metadata in bundles containing derived data must reference the input bundle
+versions. Administrators are responsible for tracking down data bundles with
+data derived from data in deleted bundles and to issue deletion requests for
+them.
+
+### Tiered storage
+
+Physical deletion of blobs is effective across all tiered storage classes in
+the underlying blob store. Additional cost may be incurred for deleting data recently moved to a
+slower access storage tier. Deletion occur There is no additional delays cause when deleting from a 
+tiered storage system such as AWS glacier.
+
+### Undoing logical deletions
+
+A bundle can be restored using `restore_bundle.py <uuid.<version>` if has not been physically deleted. This process
+uses the info from the deletion table to removes tombstones, and prevent permanent deletion. Subscribers of new 
+bundles will receive a notification that a new bundle has arrived.
+
+Deletion Table Entry Example:
+
+|Bundle.Version|admin|reason|AWS Deletion Markers (key,ID)|GCP Previous Generations (key, previous generation)|
+|--------------|-----|------|--------------------|------------------------|
+|1234-2134-3145|admin@email.com|consent|(file.obj, 1234), ...| (file.obj, 4321), ...|
+
+A bundle is restored in AWS by deleting the deletion marker and the tombstone for each files in the bundle.
+In GCS a bundle can be restored by by storing a copy of the orginal files in the bucket with the same name, and deleting 
+the tombstones.
+
+### Resources
+
+* [AWS S3 Versioning](https://docs.aws.amazon.com/AmazonS3/latest/dev/Versioning.html),
+* [AWS S3 Deleting Object Versions](https://docs.aws.amazon.com/AmazonS3/latest/dev/DeletingObjectVersions.html),
+* [AWS S3 Restoring Previous Versions](https://docs.aws.amazon.com/AmazonS3/latest/dev/RestoringPreviousVersions.html),
+* [AWS Object Lifecycle Management](https://docs.aws.amazon.com/AmazonS3/latest/dev/object-lifecycle-mgmt.html),
+* [Amazon DynamoDB TTL](https://aws.amazon.com/about-aws/whats-new/2017/02/amazon-dynamodb-now-supports-automatic-item-expiration-with-time-to-live-ttl/)
+* [GCS Object Versioning](https://cloud.google.com/storage/docs/object-versioning),
+* [GCS Object Lifecycle Management](https://cloud.google.com/storage/docs/lifecycle)
+* [HumanCellAtlas/data-store#1662 Explore Bucket Versioning for Deletion](https://github.com/HumanCellAtlas/data-store/issues/1662)
+* [HumanCellAtlas/data-store#1663 Deletion Daemon Design](https://github.com/HumanCellAtlas/data-store/issues/1663)
+
+### Open Issues
+
+TODO: remove support for `all-versions` tombstone in Data Store
+
+TODO: implement consistency check for dangling bundles
+
+TODO: implement object deletion and restoration for AWS anf GCP
+
+TODO: implement delete_bundle.py
+
+TODO: implement restore_bundle.py
+
+TODO: enable bucket versioning and lifecycle rules for replicas.
+
+TODO: Remove `Bundle/Delete` API from DSS.
```

Comment:
I don't agree that this should be done. As stated in my other comments here, I think the deletion system should have a well-defined API, and that API should be served over HTTP and available through the same endpoint as the rest of DSS. It should not be a maintenance script requiring infrastructure access. It should be secured using the shared auth infrastructure that we developed partly for this purpose.